include/common/core/freertos_message_queue_poller.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "freertos_message_queue.hpp"
+
+namespace freertos_message_queue_poller {
+
+/**
+ * Concept describing a handler of messages.
+ *
+ * @tparam Handler Handler type
+ * @tparam MessageType Message type
+ */
+template <class Handler, typename MessageType>
+concept MessageHandler = requires(Handler handler, const MessageType& message) {
+    {handler.handle_message(message)};
+};
+
+/**
+ * A FreeRTOSTask entry point that polls a MessageQueue and notifies a Handler
+ * with new messages.
+ *
+ * @tparam Message Message type
+ * @tparam Handler Handler Type
+ */
+template <typename Message, typename Handler>
+requires MessageHandler<Handler, Message>
+class FreeRTOSMessageQueuePoller {
```

Comment:
bit of a nit but this isn't really and doesn't really need to be a poller, it's fully reactive I think